Provide a `pause()` helper to eachMessage/eachBatch (#1364)
tulios/kafkajs: docs/Consuming.md

* `uncommittedOffsets()` returns all offsets by topic-partition which have not yet been committed.
* `isRunning()` returns true if consumer is in running state, else it returns false.
* `isStale()` returns whether the messages in the batch have been rendered stale through some other operation and should be discarded. For example, when calling [`consumer.seek`](#seek) the messages in the batch should be discarded, as they are not at the offset we seeked to.

> * `pause(optionalTimeout)` can be used to pause the consumer for the current topic/partition for an optional period of time. All offsets resolved up to that point will be committed (subject to [autoCommit](#auto-commit) settings) and processing of this batch will be aborted. If no timeout is specified, the consumer will be paused indefinitely until `consumer.resume()` is called with the appropriate topic/partition. See [Pause & Resyume](#pause-resume) for more information about this feature.

Reviewer: ```suggestion
* `pause(optionalTimeout)` can be used to pause the consumer for the current topic/partition for an optional period of time. All offsets resolved up to that point will be committed (subject to [autoCommit](#auto-commit) settings) and processing of this batch will be aborted. If no timeout is specified, the consumer will be paused indefinitely until `consumer.resume()` is called with the appropriate topic/partition. See [Pause & Resume](#pause-resume) for more information about this feature.
```

Typo

* `uncommittedOffsets()` returns all offsets by topic-partition which have not yet been committed.
* `isRunning()` returns true if consumer is in running state, else it returns false.
* `isStale()` returns whether the messages in the batch have been rendered stale through some other operation and should be discarded. For example, when calling [`consumer.seek`](#seek) the messages in the batch should be discarded, as they are not at the offset we seeked to.

> * `pause()` can be used to pause the consumer for the current topic/partition. All offsets resolved up to that point will be committed (subject to `eachBatchAutoResolve` and [autoCommit](#auto-commit) settings). If you want to pause in the middle of the batch, you should throw an exception after pausing if you do not wish to resolve the current batch's offset or, alternatively, make sure that `eachBatchAutoResolve` is turned off. You can used the returned callback function to resume processing of the relevant partition and topic (i.e. using `setTimeout` or some other asynchronous trigger). See [Pause & Resume](#pause-resume) for more information about this feature.

Reviewer: ```suggestion
* `pause()` can be used to pause the consumer for the current topic/partition. All offsets resolved up to that point will be committed (subject to `eachBatchAutoResolve` and [autoCommit](#auto-commit) settings). If you want to pause in the middle of the batch without resolving the current offset, you should throw an error. Alternatively, make sure that `eachBatchAutoResolve` is turned off. You can used the returned callback function to resume processing of the relevant partition and topic (i.e. using `setTimeout` or some other asynchronous trigger). See [Pause & Resume](#pause-resume) for more information about this feature.
```

})
```

> As a convenience, the `eachMessage` callback provides a simple callback function to pause the specific topic/partition for the message currently being processed. You can also specify a timeout for when that topic/partition will automatically be resumed:

Reviewer: ```suggestion
As a convenience, the `eachMessage` callback provides a `pause` function to pause the specific topic-partition of the message currently being processed.
```
